### PR TITLE
feat(perf): Improve perf of the ObjectAnimationUsingKeyFrame

### PR DESCRIPTION
**Pull request by dr1rrb**
fixes https://github.com/unoplatform/uno/issues/7072

## Performance
Improve perf of the `ObjectAnimationUsingKeyFrame`

## What is the current behavior?
All frames of the `ObjectAnimationUsingKeyFrame` are scheduled on start, and cancelled if the animation is stop before the end

## What is the new behavior?
Frames are scheduled one by one

## PR Checklist
- [ ] ~~Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)~~
- [x] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [x] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [x] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

### CODE REVIEW

File: src/Uno.UI/UI/Xaml/Media/Animation/KeyFrameScheduler.cs [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+			IEnumerable<IKeyFrame<TValue>>? frames,
+			OnFrame onFrame,
+			Action<EndReason> onCompleted)
+		{
+			CurrentValue = initialValue;
+
+			frames ??= Enumerable.Empty<IKeyFrame<TValue>>();
+			frames = frames.Trim();
+			frames = duration.HasValue
+				? frames.Where(k => k != null && k.KeyTime.TimeSpan <= duration.Value)
```
Comment (Youssef1313) [inline src/Uno.UI/UI/Xaml/Media/Animation/KeyFrameScheduler.cs:71]: nit: Can `k` be null here? We do `.Trim()` above.

File: src/Uno.UI/UI/Xaml/Media/Animation/KeyFrameScheduler.cs [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+				: frames.Trim();
```
Comment (Youssef1313) [inline src/Uno.UI/UI/Xaml/Media/Animation/KeyFrameScheduler.cs:72]: We already do `frames.Trim()` above. Isn't this one redundant?

File: src/Uno.UI/UI/Xaml/Media/Animation/KeyFrameScheduler.cs (hunk 1 above)
Comment (dr1rrb) [reply src/Uno.UI/UI/Xaml/Media/Animation/KeyFrameScheduler.cs:71]: Good catch, I lost it while playing with git, but I've removed the `.Trim()` above to reduce the number of operators (a bit more performant)

File: src/Uno.UI/UI/Xaml/Media/Animation/KeyFrameScheduler.cs [hunk 3]
```diff
@@ -0,0 +1,258 @@
...
+			{
+				RunNextFrame();
+			}
+		}
+
+		private void ScheduleNextFrame(TimeSpan delay)
+		{
+			if (_timer is null)
+			{
+				_timer = _timer = DispatcherQueue.GetForCurrentThread().CreateTimer();
```
Comment (jeromelaban) [inline src/Uno.UI/UI/Xaml/Media/Animation/KeyFrameScheduler.cs:167]:
```suggestion
				_timer = DispatcherQueue.GetForCurrentThread().CreateTimer();
```

**Review by jeromelaban — APPROVED** [review]

**Comment by MartinZikmund** [comment]
@dr1rrb Diffignore sacrifice needed 😀

**Pull request merged**